widgetssdk/src/main/java/com/glia/widgets/core/visitor/GliaVisitorMediaRepository.java

```diff
@@ -37,41 +43,135 @@ public void onEngagementEnded(Engagement engagement) {
 
     public void addVisitorMediaStateListener(VisitorMediaUpdatesListener listener) {
         visitorMediaUpdatesListeners.add(listener);
-        updateVisitorMediaState(currentMediaState);
+        notifyVisitorMediaStateChanged(currentMediaState);
         notifyOnHoldStateChange(isOnHold);
     }
 
     public void removeVisitorMediaStateListener(VisitorMediaUpdatesListener listener) {
         visitorMediaUpdatesListeners.remove(listener);
     }
 
-    private void updateVisitorMediaState(VisitorMediaState state) {
-        currentMediaState = state;
-        visitorMediaUpdatesListeners.forEach(listener -> listener.onNewVisitorMediaState(currentMediaState));
+    public Completable muteVisitorAudio() {
+        currentMediaState.getAudio().mute();
+        notifyVisitorMediaStateChanged(currentMediaState);
+        return Completable.complete();
```

Comment:
in this case, more appropriate is to use `Completable.fromAction()`, that would turn any exceptions to rx errors that can be handled in single place.